# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1321863399 18000
#      Mon Nov 21 03:16:39 2011 -0500
# Node ID 19e11eb178123deb0f87ef4e0a3b2d125acfeea4
# Parent  1367f2db49a2a16ee42e600fe1fd259f57b22dd6
quote filenames passed to the shell

* ls.m: Quote filenames passed to the shell.

diff --git a/scripts/miscellaneous/ls.m b/scripts/miscellaneous/ls.m
--- a/scripts/miscellaneous/ls.m
+++ b/scripts/miscellaneous/ls.m
@@ -45,17 +45,20 @@ function retval = ls (varargin)
     ## Initialize value for __ls_command__.
     ls_command ();
   endif
 
   if (iscellstr (varargin))
 
     args = tilde_expand (varargin);
 
-    cmd = sprintf ("%s ", __ls_command__, args{:});
+    ## FIXME -- this will fail for filenames that contain single quote
+    ## characters...
+    cmd = sprintf (" '%s'", args{:});
+    cmd = sprintf ("%s%s", __ls_command__, cmd);
 
     if (page_screen_output () || nargout > 0)
 
       [status, output] = system (cmd);
 
       if (status == 0)
         if (nargout == 0)
           puts (output);
